### ISSUE TITLE
Maximum push_amt inconsistency with maximum sendpayment 

### ISSUE THREAD

**Issue by offerm** [lightningnetwork/lnd#1077]
### Background

When opening a channel, can't push max capacity to the peer.
Channel capacity is 1,000,000 SAT.

a. can't push full capacity to peer (need to leave the commit fee on my side, why???). 
b. can't push full capacity minus commit fee to peer (need to leave a balance of 1146 , why??)
c.  can't push full capacity minus commit fee minus 1146  
d. can push full capacity minus commit fee minus 1146 minus 1
e. See case #1076 : If I open the same channel without push_amount, I can later send to the peer a maximum of 978,800 sat. But with push_amt I can send 983803. This does not make sense.

See execution script below. 

### Your environment

+ lnd --version
lnd version 0.4.1-beta commit=bc029b9cd4c15ff974c4e232fb4a606e9fea4f8c
+ uname -a
Darwin admins-MacBook-Pro-2.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
+ btcd --version
btcd version 0.12.0-beta

### Steps to reproduce

Easy, follow the execution script below.
I can send a bash script is needed. 

### Expected behaviour

Tell us what should happen

### Actual behaviour

admins-MacBook-Pro-2:dev admin$ bash -x ./test2.bash 
+ shopt -s expand_aliases
+ source /Users/admin/.bashrc
++ alias 'lncli-alice=lncli --rpcserver=localhost:10001 --no-macaroons'
++ alias 'lncli-bob=lncli --rpcserver=localhost:10002 --no-macaroons'
++ alias 'lncli-charlie=lncli --rpcserver=localhost:10003 --no-macaroons'
++ alias 'mineBlocks=btcctl --simnet --rpcuser=kek --rpcpass=kek generate '
+ lnd --version
lnd version 0.4.1-beta commit=bc029b9cd4c15ff974c4e232fb4a606e9fea4f8c
+ uname -a
Darwin admins-MacBook-Pro-2.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
+ btcd --version
btcd version 0.12.0-beta
++ lncli --rpcserver=localhost:10001 --no-macaroons listchannels
++ grep remote_pubkey
++ cut -d '"' -f 4
+ bob_pub='023c27e93757937767fb6c85893e8288cbe8735a6ef154a4db283e2519fb29d5e1
023c27e93757937767fb6c85893e8288cbe8735a6ef154a4db283e2519fb29d5e1
023c27e93757937767fb6c85893e8288cbe8735a6ef154a4db283e2519fb29d5e1'
+ btcctl --simnet --rpcuser=kek --rpcpass=kek generate 6
[
  "0098b381e0bc59f328c680ee736f557c07b822d44c0f1efd9d82cb3dcd09c0cb",
  "0143f7bb2ac7ceb40d2c85bc8d703053ea3e3771d14bf26d07731e77ea8d3d77",
  "0064593288b49dc34b12ba2d45970a3e81edfefbf8d3cf735c4e5c52b4745741",
  "01b91523ae6105deeab171409261fba04517fc4ea3aeac97806fbaf3805ca66c",
  "00456a6c3c310cd5f58796590d0a91056578d750cf2e983553897640fbfff1a2",
  "007fa905c86d0ffa2cce15f3afe449f657eadc02ed42be8e40994e943f74b6b1"
]
**+ lncli --rpcserver=localhost:10001 --no-macaroons openchannel --node_key=023c27e93757937767fb6c85893e8288cbe8735a6ef154a4db283e2519fb29d5e1 --local_amt=1000000 --push_amt=990950**
[lncli] rpc error: code = Unknown desc = **Funder balance too small (0) with fee=9050 sat, minimum=1146 sat required**
+ **lncli --rpcserver=localhost:10001 --no-macaroons openchannel --node_key=023c27e93757937767fb6c85893e8288cbe8735a6ef154a4db283e2519fb29d5e1 --local_amt=1000000 --push_amt=989804**
[lncli] rpc error: code = Unknown desc = **Funder balance too small (1146) with fee=9050 sat, minimum=1146 sat required**
+ **lncli --rpcserver=localhost:10001 --no-macaroons openchannel --node_key=023c27e93757937767fb6c85893e8288cbe8735a6ef154a4db283e2519fb29d5e1 --local_amt=1000000 --push_amt=989803**
{
	"**funding_txid": "b774eef3421128124d0c1be86d3a47e2eb13e4bc7acc3466f1e2e8ac37379013"
}**
+ btcctl --simnet --rpcuser=kek --rpcpass=kek generate 6
[
  "010c83794a25b0c7bcf11ff2e35973046ebb592cefa1fe0154676f0cc21dfe4a",
  "0027eaf7518c4a0f18182d04eae7e0d65d819496772c9425edfef403eeb1fba8",
  "00e90a715ca4581f72490787bf2503a78329c5b61775ce1c53e3fadd721e0e3a",
  "019e4afb02747a7e0f9c3698a1216a5c2f4d00ea716d6610a08530089ed45b4a",
  "0111e4f297f46a4d79548ad19136e4f1f58fccf5fde259717971b006ae7ac6d9",
  "01d7901c17c76423f90a5b23b471f5730562b7129c162983f8b540f3e009cf5d"
]
admins-MacBook-Pro-2:dev admin$

**Comment by Roasbeef** [comment]
Yes, we do not allow pushing the entire capacity to the remote mode for opening channels. This is due to the fact that the initiator needs to pay fees on the commitment transaction. If the entire amount is pushed, then neither party is available to pay for the transaction fees.

**Comment by offerm** [comment]
Summary: I can open 1,000,000 sat channel and use a maximum push_amt of 989803 sat.

There are two problems here:

1. small - according to the error message 

> under balance too small (0) with fee=9050 sat, minimum=1146 sat required

I should be able to push_amt 989,804 (leaving 1146) sat. but when doing so I get 

> Funder balance too small (1146) with fee=9050 sat, minimum=1146 sat required

(small code issue)

2. **bigger** - taking the commit_fee and reserve (1%) into account, I should be able to push a maximum of 980950 sat (1,000,000 - 9,050 - 10,000) but I manage to push 989803.  One may claim that the reserve is not relevant when opening the transaction, in that case the maximum amount should be 990,950 sat and not 989804.

**Comment by Roasbeef** [comment]
The reserve _is_ relevant, as we still require skin in the game to exist at that first state since things are fully unidirectional. If_all_ funds are pushed over, then no one is able to pay for commitment fees.

**Comment by offerm** [comment]
So, for funding transaction that pushes amt to the remote peer, the reserve is not being considered. This is in line with the spec (BOLT 2):

> The sender can unconditionally give initial funds to the receiver using a non-zero push_msat — this is one case where the normal reserve mechanism doesn't apply. 

The code is calculating the maximum amount that can be pushed by 
`ourBalance = 1,000,000 - commitment_fee (9,050) - push_amount
`
once we have our balance the code checks:
```
       if initiator && ourBalance.ToSatoshis() <= 2*DefaultDustLimit() {
		return nil, ErrFunderBalanceDust(
			int64(commitFee),
			int64(ourBalance.ToSatoshis()),
			int64(2*DefaultDustLimit()),
		)
	}
```

1146 is 2*DefaultDustLimit().

Hence, maximum push_amt is 1,000,000 - 9050 - 1146 = 989,804.

Still, when I try to push_amt=989,804 (leaving 1146) sat, I get

> Funder balance too small (1146) with fee=9050 sat, minimum=1146 sat required

**Which is, IMHO,  a bug.**

**Comment by Roasbeef** [comment]
The check is a `<=`, so if you only have `1146` satoshis left, we reject that. I'm failing to see the bug here. The one improvement that could be made is modifying the error message a bit to make this clearer. 


Note the interaction with fees, if you only have `1146` satoshis, it's likely the case that the _other_ party can't even add a series of HTLC's, as you don't have enough balance left in the channel to pay for those fees.

**Comment by offerm** [comment]
Let me try to explain again:

When I try to push_amt=989804 I get the following error

> [lncli] rpc error: code = Unknown desc = Funder balance too small (**1146**) with fee=9050 sat, minimum=**1146** sat required

Basically it says that post action the funder balance will be 1146 sat BUT it should have a minimum of 1146 sat. 
Since the balance is equal to the minimum this should not be an error.

The cause is exactly in the `<=` check. It should be `<`.

Even the comment above the check is calling for `<` test. 
```
        // If we're the initiator and our starting balance within the channel
	// after we take account of fees **is below 2x** the dust limit, then we'll
	// reject this channel creation request.
	//
	// TODO(roasbeef): reject if 30% goes to fees? dust channel
	if initiator && ourBalance.ToSatoshis() <= 2*DefaultDustLimit() {
```

IMHO, This should change to `<`.

**Comment by Roasbeef** [comment]
> Let me try to explain again:

I understand what you're reporting. You're arbitrarily assuming that the check should be modified to allow an amount _directly_ at 2x the dust limit. However, we don't want to even create outputs that small, as they likely won't be sweepable on-chain. Instead, the error message just needs to be updated to reflect the new absolute amount. 

We should likely even raise the amount to far beyond 2x dust at this point.
